### PR TITLE
[onert] Use structured binding for pair

### PR DESCRIPTION
**Pull request by ragmani**
This commit applies structured binding to pair.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: runtime/onert/core/src/compiler/StaticShapeInferer.cc [hunk 1]
```diff
@@ -202,8 +202,11 @@ StaticShapeInferer::createStaticShapeInferers(
   }
 
   // Append observers in all StaticShapeInferers
-  for (auto &&[subg_index, lowered_subg] : lowered_subgs)
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/StaticShapeInferer.cc]: Structured bindings can be captured by lambda expressions since c++20.

**Comment by jyoungyun** [comment]
@ragmani LGTM Please resolve conflict issue.

File: runtime/onert/core/src/exec/JSONExecTime.cc [hunk 2]
```diff
@@ -135,6 +135,9 @@ void JSON::printOperation(const std::map<uint32_t, int64_t> &operation_info,
   stream.seekp(-2, std::ofstream::end);
 }
 
+using MeasurementData = std::unordered_map<
+  const backend::Backend *,
+  std::unordered_map<std::string, std::unordered_map<bool, std::map<uint32_t, int64_t>>>>;
```
Comment (glistening) [inline runtime/onert/core/src/exec/JSONExecTime.cc:140]: This seems not relevant to this PR. Furthermore, `MeasurementData` seems not used.

**Review by glistening — DISMISSED** [review]
LGTM except for https://github.com/Samsung/ONE/pull/13499#discussion_r1692375028

File: runtime/onert/core/src/exec/JSONExecTime.cc (hunk 2 above)
Comment (ragmani) [reply runtime/onert/core/src/exec/JSONExecTime.cc:140]: You are right. I forgot to remove it. Thanks for your pointing out.

**Review by glistening — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]
LGTM

**Pull request merged**